src/main/scala/io/iohk/ethereum/vm/UInt256.scala

```diff
@@ -0,0 +1,196 @@
+package io.iohk.ethereum.vm
+
+import akka.util.ByteString
+import language.implicitConversions
+
+// scalastyle:off number.of.methods
+object UInt256 {
+
+
+  /** Size of UInt256 byte representation */
+  val Size: Int = 32
+
+  private val Modulus: BigInt = BigInt(2).pow(256)
+
+  val MaxValue: UInt256 = new UInt256(Modulus - 1)
+
+  val Zero: UInt256 = new UInt256(0)
+
+  val One: UInt256 = new UInt256(1)
+
+  val Two: UInt256 = new UInt256(2)
+
+
+  def apply(bytes: ByteString): UInt256 = {
+    require(bytes.length <= Size, s"Input byte array cannot be longer than $Size: ${bytes.length}")
+    UInt256(bytes.foldLeft(BigInt(0)){(n, b) => (n << 8) + (b & 0xff)})
+  }
+
+  def apply(array: Array[Byte]): UInt256 =
+    UInt256(ByteString(array))
+
+  def apply(n: BigInt): UInt256 =
+    new UInt256(boundBigInt(n))
+
+  def apply(b: Boolean): UInt256 =
+    if (b) One else Zero
+
+  def apply[N: Integral](n: N): UInt256 = {
+    val num = implicitly[Integral[N]]
+    apply(BigInt(num.toLong(n)))
```

Comment:
Not sure, we actually don't even need this one, as all the other standard numeric types are smaller than `BigInt` and will be implicitly converted. 

It could make a difference when we move away from BigInt-based implementation. But we could still change this to `def apply(n: Long): UInt256` and all smaller types (`Int`, `Short`, `Byte`) will be handled. I think I'd go with this one. What do you think?